sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -674,6 +740,65 @@ class ResolveSessionCatalog(
       comment = comment)
   }
 
+  private def buildHiveCatalogTable(
+      table: TableIdentifier,
+      schema: StructType,
+      partitioning: Seq[Transform],
+      bucketSpec: Option[BucketSpec],
+      properties: Map[String, String],
+      serdeInfo: SerdeInfo,
+      options: Map[String, String],
+      location: Option[String],
+      comment: Option[String],
+      external: Boolean): CatalogTable = {
+    val defaultStorage = HiveSerDe.getDefaultStorage(conf)
+    val baseStorage = defaultStorage.copy(
+      locationUri = location.map(CatalogUtils.stringToURI),
+      serde = serdeInfo.serde.orElse(defaultStorage.serde),
+      properties = options ++ serdeInfo.serdeProperties)
+
+    val storage = (serdeInfo.storedAs, serdeInfo.formatClasses) match {
+      case (Some(format), None) =>
+        HiveSerDe.sourceToSerDe(format) match {
+          case Some(hiveSerDe) =>
+            baseStorage.copy(
+              inputFormat = hiveSerDe.inputFormat,
+              outputFormat = hiveSerDe.outputFormat,
+              serde = serdeInfo.serde.orElse(hiveSerDe.serde))
```

Comment:
I believe the logic should be: use specified serde, or use the serde inferred from STORED AS, or use the default serde from `HiveSerDe.getDefaultStorage`